### PR TITLE
feat: support L1 data fee

### PR DESCRIPTION
**Pull request by GeekArthur**
This PR is to support OP stack L1 data fee with the following features:
* leverage Optimism gas price oracle contract to calculate L1 data fee
* support L1 data fee calculation for generic contract call, ETH transfer and ERC20 transfer
* expose OP stack support as env var for more flexibility
* leave GetL1DataFee() as interface to let chain specific Rosetta implement

### CODE REVIEW

File: services/construction/construction_service.go [hunk 1]
```diff
@@ -52,11 +54,24 @@ func NewAPIService(
...
 	return &APIService{
-		config: cfg,
-		types:  types,
-		errors: errors,
-		client: client,
+		config:    cfg,
+		types:     types,
+		errors:    errors,
+		client:    client,
+		ethClient: ethClient,
```
Comment (xiaying-peng) [inline services/construction/construction_service.go:74]: I understand what you are doing, but this is not a very optimal solution, you are passing an ethClient only used it in a contract call.
Maybe we should implement an interface in `Client`, and have you op-stack chain implement this logic? Or any other alternatives?

Comment (GeekArthur) [reply services/construction/construction_service.go]: eth client is removed, we can leverage the existing eth client now!

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.16
 
 require (
 	github.com/coinbase/rosetta-sdk-go v0.8.2
+	github.com/ethereum-optimism/optimism/op-bindings v0.10.14
```
Comment (xiaying-peng) [inline go.mod:7]: We cannot add this dependency for all non op chains

Comment (GeekArthur) [reply go.mod]: remove it and use expose as interface instead

**Review by xiaying-peng — APPROVED** [review]

**Pull request merged**